android/pytorch_android/src/main/java/org/pytorch/Tensor.java

```diff
@@ -29,55 +33,107 @@
   }
 
+  /**
+   * Creates new Tensor instance with elements type float and specified dimensions and data as
+   * java float array.
+   * Content of that java float array will be copied into newly allocated direct buffer
+   * {@link Tensor#allocateFloatBuffer(int)}.
+   *
+   * @param dims Tensor dimensions
+   * @param data Tensor elements.
+   */
   public static Tensor newFloatTensor(long[] dims, float[] data) {
     checkArgument(data != null, ERROR_MSG_DATA_ARRAY_NOT_NULL);
     checkArgument(dims != null, ERROR_MSG_DIMS_NOT_NULL);
     checkDims(dims);
     checkDimsAndDataCapacityConsistency(data.length, dims);
-    final int bufferCapacity = (int) numElements(dims);
-    final FloatBuffer floatBuffer = allocateFloatBuffer(bufferCapacity);
+    final FloatBuffer floatBuffer = allocateFloatBuffer((int) numElements(dims));
     floatBuffer.put(data);
     return new Tensor_float32(floatBuffer, dims);
   }
 
+  /**
+   * Creates new Tensor instance with elements type float and specified dimensions and data as
+   * java int array.
+   * Content of that java int array will be copied into newly allocated direct buffer
+   * {@link Tensor#allocateIntBuffer(int)}.
+   *
+   * @param dims Tensor dimensions
+   * @param data Tensor elements.
+   */
   public static Tensor newIntTensor(long[] dims, int[] data) {
     checkArgument(data != null, ERROR_MSG_DATA_ARRAY_NOT_NULL);
     checkArgument(dims != null, ERROR_MSG_DIMS_NOT_NULL);
     checkDims(dims);
     checkDimsAndDataCapacityConsistency(data.length, dims);
-    final int bufferCapacity = (int) numElements(dims);
-    final IntBuffer intBuffer = allocateIntBuffer(bufferCapacity);
+    final IntBuffer intBuffer = allocateIntBuffer((int) numElements(dims));
     intBuffer.put(data);
     return new Tensor_int32(intBuffer, dims);
   }
 
+  /**
+   * Creates new Tensor instance with elements type byte and specified dimensions and data as
+   * java byte array.
+   * Content of that java byte array will be copied into newly allocated direct buffer
+   * {@link Tensor#allocateByteBuffer(int)}.
+   *
+   * @param dims Tensor dimensions
+   * @param data Tensor elements.
+   */
   public static Tensor newByteTensor(long[] dims, byte[] data) {
     checkArgument(data != null, ERROR_MSG_DATA_ARRAY_NOT_NULL);
     checkArgument(dims != null, ERROR_MSG_DIMS_NOT_NULL);
     checkDims(dims);
     checkDimsAndDataCapacityConsistency(data.length, dims);
-    final int bufferCapacity = (int) numElements(dims);
-    final ByteBuffer byteBuffer = allocateByteBuffer(bufferCapacity);
+    final ByteBuffer byteBuffer = allocateByteBuffer((int) numElements(dims));
     byteBuffer.put(data);
     return new Tensor_byte(byteBuffer, dims);
   }
 
+  /**
+   * Creates new Tensor instance with elements type float and specified dimensions and data.
+   *
+   * @param dims Tensor dimensions, must be not negative.
+   * @param data Direct buffer with native byte order that contains corresponsing to dimensions
```

Comment:
Direct buffer with native byte order that contains {@code Tensor.numElements(dims)} elements.  The buffer is used directly without copying, and changes to its contents will change the tensor.